test/notifications.js

```diff
@@ -0,0 +1,85 @@
+var tesseract = require('../src/Tesseract.js');
+
+// Tests for listening to notifications.
+
+function connectAndRun(test, body) {
+    test.expect(2);
+
+    tesseract.connect(null, function (err, client) {
+        test.equals(err, null);
+        body(client);
+    });
+}
+
+var insertInterval;
+
+function fireInserts(client) {
+    insertInterval = setInterval(function () {
+        client.insert('people', {
+            'name': 'Joe Bloggs'
+        });
+    }, 100);
+}
+
+exports.testListenToAll = function(test) {
```

Comment:
Travis will kill a build if nothing is printed for 10 minutes. The reason is because it's stalled, these tests make sure the notifications are working which requires setting them up and waiting for them. There is a gremlin in the system that's causing something somewhere to go wrong (silently) so javascript waits for a notification that never comes... This is a bug, but only occurs in some cases. So I have to keep fiddling with it until it works 100% of the time.